src/components/rangeslider/range_plot.js

```diff
@@ -0,0 +1,161 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Symbols = require('../drawing/symbol_defs');
+var Drawing = require('../drawing');
+
+var helpers = require('./helpers');
+var dataProcessors = require('./data_processors');
+var svgNS = 'http://www.w3.org/2000/svg';
+
+module.exports = function rangePlot(gd, w, h) {
+
+    var traces = gd._fullData,
+        xaxis = gd._fullLayout.xaxis,
+        yaxis = gd._fullLayout.yaxis,
+        minX = xaxis.range[0],
+        maxX = xaxis.range[1],
+        minY = yaxis.range[0],
+        maxY = yaxis.range[1];
+
+
+    // create elements for plot and its clip
+    var clipPath = document.createElementNS(svgNS, 'path');
+    clipPath.setAttribute('d', ['M0,0', w + ',0', w + ',' + h, '0,' + h, 'Z'].join(' '));
+
+    var clip = document.createElementNS(svgNS, 'clipPath');
+    clip.setAttribute('id', 'range-clip-path');
+    clip.appendChild(clipPath);
+
+    var rangePlot = document.createElementNS(svgNS, 'g');
+    rangePlot.setAttribute('clip-path', 'url(#range-clip-path)');
+    rangePlot.appendChild(clip);
+
+
+    var processX = dataProcessors[gd._fullLayout.xaxis.type || 'category'],
+        processY = dataProcessors[gd._fullLayout.yaxis.type || 'category'];
+
+
+    // for now, only scatter traces are supported
+    var allowedTypes = ['scatter', 'scattergl'];
+
+    for(var i = 0; i < traces.length; i++) {
+
+        var trace = traces[i],
+            pointPairs = [];
+
+        if(allowedTypes.indexOf(trace.type) < 0) {
```

Comment:
Instead of a hard-coded list of support trace types, I think we should make the `makeScatter` function below a method of `Scatter` and `ScatterGl` (e.g. named `Scatter.sliderPoints`) and have this loop check for the existence of `trace._module.sliderPoints`.